File: apollo/src/main/java/io/muun/apollo/domain/errors/EmptyFieldError.java
package io.muun.apollo.domain.errors;


import io.muun.apollo.external.UserFacingErrorMessages;

public class EmptyFieldError extends UserFacingError {

    public enum Field {
        FIRST_NAME,
        LAST_NAME,
        PASSWORD
    }

    public EmptyFieldError(Field field) {
        super(UserFacingErrorMessages.INSTANCE.emptyField(field));
    }
}

File: apollo/src/main/java/io/muun/apollo/domain/errors/EmptyFieldError.kt
package io.muun.apollo.domain.errors


import io.muun.apollo.external.UserFacingErrorMessages

class EmptyFieldError(field: Field):
    UserFacingError(UserFacingErrorMessages.INSTANCE.emptyField(field)) {

    enum class Field {
        FIRST_NAME,
        LAST_NAME,
        PASSWORD
    }
}
